great_expectations/checkpoint/checkpoint.py

```diff
@@ -107,7 +112,28 @@ def get_substituted_config(
                 # don't replace _substituted_config if already exists
                 if self._substituted_config is None:
                     self._substituted_config = substituted_config
-                return substituted_config
+        return self._substitute_config_variables(config=substituted_config)
+
+    def _substitute_config_variables(
+        self, config: CheckpointConfig
+    ) -> CheckpointConfig:
+        substituted_config_variables = substitute_all_config_variables(
+            self.data_context.config_variables,
+            dict(os.environ),
+            self.data_context.DOLLAR_SIGN_ESCAPE_STRING,
+        )
+
+        substitutions = {
+            **substituted_config_variables,
+            **dict(os.environ),
+            **self.data_context.runtime_environment,
+        }
+
+        return CheckpointConfig(
+            **substitute_all_config_variables(
+                config, substitutions, self.data_context.DOLLAR_SIGN_ESCAPE_STRING
```

Comment:
Minor: If this is a constant, shouldn't it be `BaseDataContext. DOLLAR_SIGN_ESCAPE_STRING`?

Reply:
Good point - thanks for always helping me make my code better! I suppose I used `self.data_context` since in DataContext, `self.DOLLAR_SIGN_ESCAPE_STRING` is used, but now I see that that makes sense because same inheritance chain. Will change.